### ISSUE TITLE
💅 `useConsistentArrowReturn` doesn't handle awaits in Promise<void> functions

### ISSUE THREAD

**Issue by jpike88** [biomejs/biome#7666]
### Environment information

```bash
Version: 2.2.5
```

### Rule name

useConsistentArrowReturn

### Playground link

https://biomejs.dev/playground/?lintRules=useConsistentArrowReturn&code=YwBvAG4AcwB0ACAAbQB5AFAAcgBvAG0AaQBzAGUAIAA9ACAAYQBzAHkAbgBjACAAZgB1AG4AYwB0AGkAbwBuACAAKAApAHsACgAgACAAcgBlAHQAdQByAG4AIABuAGUAdwAgAFAAcgBvAG0AaQBzAGUAKAApADsACgB9AAoACgBjAG8AbgBzAHQAIAB4ACAAPQAgAGEAcwB5AG4AYwAgACgAKQA6AFAAcgBvAG0AaQBzAGUAPAB2AG8AaQBkAD4AIAA9AD4AIABtAHkAUAByAG8AbQBpAHMAZQAoACkAOwAKAAoACgBjAG8AbgBzAHQAIAB5ACAAPQAgAGEAcwB5AG4AYwAgACgAKQA6AFAAcgBvAG0AaQBzAGUAPAB2AG8AaQBkAD4AIAA9AD4AIAB7AAoAIAAgAGEAdwBhAGkAdAAgAG0AeQBQAHIAbwBtAGkAcwBlACgAKQA7AAoAfQA%3D&ruleDomains.project=all

### Expected result

It's a bit of an edge case, but the second function (variable `y`) is technically returning Promise<void> as well, and should be able to be collapsed using the safe fix that got applied to `x`.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by kedevked** [comment]
I tried to see how ESLINT handles it. And `await` without `return` is not considered as you can see it here https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgYXJyb3ctYm9keS1zdHlsZTogW1wiZXJyb3JcIiwgXCJhcy1uZWVkZWRcIl0qL1xuXG5jb25zdCBteVByb21pc2UgPSBhc3luYyBmdW5jdGlvbiAoKXtcbiAgcmV0dXJuIG5ldyBQcm9taXNlKCk7XG59XG5cbmNvbnN0IHggPSAoKT0+IG15UHJvbWlzZSgpO1xuXG5cbmNvbnN0IHkgPSBhc3luYyAoKSA9PiB7XG4gICBhd2FpdCBteVByb21pc2UoKTtcbn0iLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=

The rule is more about returned expressions if I am not mistaken

**Comment by arendjr** [comment]
Yeah, I think "a bit of an edge case" is quite the understatement here 😅

Also, it would require type information to know when this suggested fix is safe or not, because if `myPromise()` returns any `Promise<T>` that is not `Promise<void>`, it would technically become a semantically incorrect fix.

I don't see this being picked up anytime soon.
